python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -482,6 +503,24 @@ class NearestNeighbors(Base):
             When set to True, the class self.output_type is overwritten
             and this method returns the output as a cumlarray
 
+        two_pass_precision : bool, optional (default = False)
+            When set to True, a slow second pass will be used to improve the
+            precision of results returned for searches using L2-derived
+            metrics. FAISS uses the Euclidean distance decomposition trick to
+            compute distances in this case, which may result in numerical
+            errors for certain data. In particular, when several samples
+            are close to the query sample (relative to typical inter-sample
+            distances), numerical instability may cause the computed distance
+            between the query and itself to be sufficiently non-zero that it is
+            no longer returned as the first result. If this flag is set to
+            true, distances to the query vectors will be recomputed with high
+            precision for all retrieved samples, and the results will be
+            resorted accordingly. Note that for large values of k or large
+            numbers of query vectors, this correction becomes impractical in
+            terms of both runtime and memory. It should be used with care and
+            only when strictly necessary (when precise results are critical and
+            samples may be tightly clustered).
+
```

Comment:
We duplicate the documentation for the other parameters, and this seems fairly relevant to developers trying to understand the details and purpose of this flag. Frankly, it probably will only be used for practical purposes by people whose applications require a fairly low-level understanding of the algorithms invoked, so it seems relevant to both developers and consumers. I know we've discussed ways of passing through document strings for parameters like this to avoid having them go out of sync, but I'm not sure I've seen us do that anywhere else in the codebase. Do you know if we have some tool for that at this point?